tests/integration/test_cloud_storage_api_client.py

```diff
@@ -0,0 +1,91 @@
+    
+    1. .store() and .fetch() are inverse operations, and then .delete() cleans up after .store()
+    2. Testing these functions separately requires independently recreating the integrations,
+    3. Which duplicates the logic while failing to actually isolate the integration behaviors.
+    """
+    # Given an artifact store and a file containing Ada Lovelace's name:
+
+    storage_client = CloudStorageAPIClient(project_name="mozdata", bucket_name="mozdata-tmp")
+
+    string_to_store = "Ada Lovelace"
+    timestamp = datetime.now().strftime("%Y-%m-%d-%H-%M-%S")
+    filename_to_store_it_at = f"first_computer_programmer_{timestamp}.txt"
+    encoded_string = string_to_store.encode(encoding='utf-8')
+
+    # When we use .store() to call for her name to be stored on GCS, the command succeeds: 
+
+    filepath = storage_client.store(data=encoded_string, storage_path=filename_to_store_it_at)
+    assert filepath == filename_to_store_it_at, "The model was not stored as we expect."
+
+    # And when we use .fetch() to call for the file to be fetched from GCS, we can retrieve her name:
+
+    try:
+        storage_client.fetch(remote_path=filename_to_store_it_at, local_path=f"{tmp_path}/{filename_to_store_it_at}")
+
+        with open(f"{tmp_path}/{filename_to_store_it_at}", "rb") as file:
+            encoded_stored_string = file.read()
+            decoded_stored_string = encoded_stored_string.decode(encoding='utf-8')
+            assert decoded_stored_string == string_to_store, "The model was not fetched as we expect."
+
+    # And when we use .delete() to call for the file to be deleted, provided it's possible that our file _uploaded_, we can delete it: 
+    # (Lives in a finally block so the file is cleaned up even if fetching fails)
+    finally: 
+        storage_client._CloudStorageAPIClient__delete(filename_to_store_it_at)
+
+        try:
+            # This line assumes fetch is working (which we believe we're testing for above)
+            storage_client.fetch(remote_path=filename_to_store_it_at, local_path=f"{tmp_path}/{filename_to_store_it_at}")
+            
+            pytest.fail("The model was not deleted as we expect; it's still on the GCS file system.")
+        except Exception as e:
+            assert "No such object" in e.message
+
+@pytest.mark.integration
+def test_store__existing_filename__throws_clear_exception():
+    """
+    Tests that, if we try to store a file in GCS when there's already a file of that name, we get an exception that clearly explains what happened.
+
+    I added this test because this happened to me while writing another test, and I found the GCS error message cryptic.
+    """
+    # Given an artifact store and a file containing Grace Hopper's name:
+
+    storage_client = CloudStorageAPIClient(project_name="mozdata", bucket_name="mozdata-tmp")
+
+    string_to_store = "Grace Hopper"
+    identifier = os.getlogin() if os.getlogin() else "anonymous"
```

Comment:
I'm afraid `os.getlogin` has to go from this test as well for the same reasons as the previous tests (I can't currently run the tests on my machine)

Reply:
AH YES I missed that. Thank you!